core/src/main/java/org/ehcache/spi/cache/tiering/AuthoritativeTier.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.spi.cache.tiering;
+
+import org.ehcache.exceptions.CacheAccessException;
+import org.ehcache.function.Function;
+import org.ehcache.spi.cache.Store;
+
+/**
+ * Authoritative tier, according to Montreal design.
+
+ * @author Ludovic Orban
+ */
+public interface AuthoritativeTier<K, V> extends Store<K, V> {
+
+  /**
+   * Marks the mapping as not evictable and returns it atomically.
+   * @return the value holder.
+   */
+  ValueHolder<V> getAndFault(K key) throws CacheAccessException;
+
+  /**
+   * Marks the mapping as not evictable and performs computeIfAbsent() atomically.
+   * @return the value holder.
+   */
+  ValueHolder<V> computeIfAbsentAndFault(K key, Function<? super K, ? extends V> mappingFunction) throws CacheAccessException;
+
+  /**
+   * This marks the entry as evictable again.
+   * @return true if something was flushed, false otherwise.
+   */
+  boolean flush(K key, ValueHolder<V> valueHolder, CachingTier<K, V> cachingTier);
```

Comment:
It would be a lot cleaner to add the expiry stuff to the ValueHolder. Unfortunately, the way expiring vs eternal value holders are implemented in the OnHeapStore means that this would require a non-trivial amount of work, so I consider it another task that I will add to the Waffle board.

Louis made the exact same remark (in one of the outdated comments), and here is what I responded:

That's the reason why there also is a CachingTier object passed in: the ValueHolder interface doesn't expose expiration methods that are needed to copy the expiration back to the authoritative tier, hence the authoritative tier asks the caching tier for that data to copy it in its own ValueHolder.

We may want to revisit the ValueHolder interface, or at least the way I implemented this expiration-copying mechanism.